benchmarks/nemo/nemolite2d/manual_versions/regent/kind_params_mod.f90

```diff
@@ -0,0 +1,14 @@
+!> Module holding basic KIND parameters
+module kind_params_mod
+  use iso_c_binding
+  implicit none
+
+  public
+
+  !> Douple precision kind parameter
+  integer, parameter :: GO_WP = c_double !SELECTED_REAL_KIND(12,307)
```

Comment:
Is the dl_esm_inf library safe to use for the C versions without this definition btw? Should we not be using interoperable types in that anyway?

Reply:
dl_esm_inf is very Fortran-centric (like the other infrastructure libraries that PSyclone deals with and we don't have direct control over like LFRic) so it is probably OK as it is but we can consider the change if it is useful. Anyway this is probably a discussion for a separate issue.

But for Regent I see gocean2d_io_mod is an isolated copy that doesn't use the dl_esm_inf infrastructure, so I think it would be OK to replace the types directly there to use the c bindings. Alternatively, you can keep the `kind_params_mod.f90` but making clear in a top comment that this imitate the naming of dl_esm_inf types but it uses C types instead.

Reply:
Did the latter.